otter/convergence/gathering.py

```diff
@@ -84,21 +83,27 @@ def group_id(s):
     return get_all_server_details(changes_since).on(servers_apply)
 
 
-def merge_servers(first, second):
+def updated_deleted_servers(old, new):
     """
-    Return servers got by merging the two sets of servers where second
-    takes precedence over first
+    Return servers with servers not found in new list marked as deleted
 
-    :param list first: List of server dicts
-    :param list second: List of server dicts
-    :return: List of merged server dicts
+    :param list old: List of old servers
+    :param list second: List of latest servers
+    :return: List of updated servers
     """
+
     def sdict(servers):
         return {s['id']: s for s in servers}
 
-    return merge(sdict(first), sdict(second)).values()
+    old = sdict(old)
+    new = sdict(new)
+    deleted_ids = set(old.keys()) - set(new.keys())
+    for s in deleted_ids:
+        old[s]["status"] = "DELETED"
```

Comment:
I think we can get rid of this mutation *and* make the function a lot shorter by using pyrsistent's `transform`:

`return pmap(merge(old, new)).transform([deleted_ids.__contains__, "status"], "DELETED")`

This is a little tricky, but exactly the kind of use case that `transform` was meant for. Plus it's purely functional :)

(I double-checked that this should do the right thing:

```python
>>> servers = pmap({'s1': pmap(), 's2': pmap(), 's3': pmap()})
>>> deleted = ['s1', 's3']
>>> servers.transform([deleted.__contains__, "status"], "DELETED")
pmap({'s3': pmap({'status': 'DELETED'}), 's2': pmap({}), 's1': pmap({'status': 'DELETED'})})
```

Okay!)